numpy/core/src/npysort/quicksort.c.src

```diff
@@ -198,6 +208,7 @@ stack_pop:
         }
         pr = *(--sptr);
         pl = *(--sptr);
+        cdepth = *(--psdepth);
```

Comment:
Well, it has the advantage the need to increment depth on every push is made clear. I think it makes the logic a bit clearer because everything takes place in a few lines. Note that you do need the `++depth` so that the depth is strictly increasing up the stack so only the top partition can be heapsorted on any pass through.